test/validation/api/ipsec/ipsec.c

```diff
@@ -854,6 +869,37 @@ static int ipsec_send_out_one(const ipsec_test_part *part,
 	return num_out;
 }
 
+void ipsec_test_sa_update_seq_num(odp_ipsec_sa_t sa, uint32_t seq_num)
+{
+	odp_ipsec_test_sa_param_t sa_param;
+
+	sa_param.type = ODP_IPSEC_TEST_SA_SEQ_NUM;
+	sa_param.val.seq_num = seq_num;
+
+	CU_ASSERT_EQUAL(0, odp_ipsec_test_sa_update(sa, sa_param));
```

Comment:
If we allow this function to fail even when the capability call succeeds (so that the capa bit does not promise that update is possible for all SAs at all times), then it is not strictly correct to fail validation test if the call fails. Maybe we could live with the strict check now but it would be better to just notice if the update failed and then turn of the sequence number checking until the update call succeeds again.

Reply:
Agreed. Update the test to loop on the API call till either it succeeds or one minute timeout happens.

Reply:
I think one minute timeout is excessive as it may repeat for multiple algorithms and then possible slow down CI a lot. Something like one second or less would be better if retrying made sense, but I do not think it does. If the call fails, it probably fails for a reason that does not go away by waiting it out (e.g. for this particular SA seqno cannot be set through this API).

My original point was that since the API says that the function may fail, it is not a violation of the API if the function fails and thus it is not strictly correct for the validation check to fail in that case. Ideally, the validation test would notice that the function failed and then make note of that so that it won't do the subsequent checks that check the outcome of the attempted test operation (that was not done in the first place because the call failed).

Reply:
Updated the test to skip further check related to seq_num when the API call fails. Also the test would print ">> skipped" for the test configuration which was skipped.